src/main/java/com/hedera/downloader/RecordFileDownloader.java

```
@@ -151,6 +153,45 @@ public static void verifyValidRecordFiles(String validDir) throws Exception {
 		}
 	}
 
+	/**
+	 * Verify the .rcd files to see if the file Hash matches prevFileHash
+	 * @param fileToCheck
+	 * @throws Exception 
+	 */
+	public static boolean verifyValidRecordFile(File objfileToCheck) throws Exception {
+		String lastValidRecordFileName =  applicationStatus.getLastValidDownloadedRecordFileName();
+		String lastValidRecordFileHash = applicationStatus.getLastValidDownloadedRecordFileHash();
+		if (!objfileToCheck.exists()) {
```

Move fail fast if statement to be first to avoid potential db lookup.

---

Done